ooniprobe/Model/Database/Measurement.mm

```diff
@@ -96,4 +96,32 @@ -(void)deleteObject{
     [self remove];
 }
 
+-(void)getExplorerUrlSuccessCallback:(void (^)(NSString*))success errorCallback:(void (^)(NSError*))error {
+    NSLog(@"%@ getExplorerUrl",self.Id);
+    NSMutableString *path = [NSMutableString stringWithFormat:@"https://api.ooni.io/api/v1/measurements?report_id=%@",
+                      self.report_id];
+    if ([self.test_name isEqualToString:@"web_connectivity"])
+        [path appendString:[NSString stringWithFormat:@"&input=%@",
+                            self.url_id.url]];
+    NSURL *url = [NSURL URLWithString:path];
+    NSURLSessionDataTask *downloadTask =
+    [[NSURLSession sharedSession]
+     dataTaskWithURL:url
+     completionHandler:^(NSData *data, NSURLResponse *response, NSError *nserror) {
+         if (nserror != nil)
+             error(nserror);
```

Comment:
Please, don't increase the code complexity. Just return after an error.

Reply:
I don't understand what you say.
Didn't you say we need to call the error callback with the NSError object?